language/stdlib/modules/transaction_fee.move

```diff
@@ -1,105 +1,104 @@
+        preburn: Preburn<CoinType>
     }
 
-    // Called from block metadata transaction for each currency
-    public fun distribute_transaction_fees<Token>() acquires TransactionFees {
-      // Can only be invoked by LibraVM privilege.
-      Transaction::assert(Transaction::sender() == 0x0, 33);
-
-      let num_validators = LibraSystem::validator_set_size();
-      let amount_collected = LibraAccount::balance<Token>(0xFEE);
-
-      // If amount_collected == 0, this will also return early
-      if (amount_collected < num_validators) return ();
-
-      // Calculate the amount of money to be dispursed, along with the remainder.
-      let amount_to_distribute_per_validator = per_validator_distribution_amount(
-          amount_collected,
-          num_validators
-      );
+    /// We need to be able to determine if `CoinType` is LBR or not in
+    /// order to unpack it properly before burning it. This resource is
+    /// instantiated with `LBR` and published in `TransactionFee::initialize`.
+    /// We then use this to determine if the / `CoinType` is LBR in `TransactionFee::is_lbr`.
+    resource struct LBRIdent<CoinType> { }
 
-      // Iterate through the validators distributing fees equally
-      distribute_transaction_fees_internal<Token>(
-          amount_to_distribute_per_validator,
-          num_validators,
-      );
+    /// Called in genesis. Sets up the needed resources to collect
+    /// transaction fees by the `0xB1E55ED` account.
+    public fun initialize(blessed_account: &signer, fee_account: &signer) {
+        Transaction::assert(Signer::address_of(blessed_account) == 0xB1E55ED, 0);
+        let cap = LibraAccount::extract_sender_withdrawal_capability(fee_account);
+        move_to(blessed_account, TransactionFeeCollection { cap });
+        move_to(blessed_account, LBRIdent<LBR>{})
     }
 
-    // After the book keeping has been performed, this then distributes the
-    // transaction fees equally to all validators with the exception that
-    // any remainder (in the case that the number of validators does not
-    // evenly divide the transaction fee pot) is distributed to the first
-    // validator.
-    fun distribute_transaction_fees_internal<Token>(
-        amount_to_distribute_per_validator: u64,
-        num_validators: u64,
-    ) acquires TransactionFees {
-        let distribution_resource = borrow_global<TransactionFees>(0xFEE);
-        let index = 0;
-
-        while (index < num_validators) {
+    /// Sets ups the needed transaction fee state for a given `CoinType`
+    /// currency.
+    public fun add_txn_fee_currency<CoinType>(fee_account: &signer, burn_cap: &BurnCapability<CoinType>) {
+        Transaction::assert(Signer::address_of(fee_account) == 0xFEE, 0);
+        LibraAccount::add_currency<CoinType>(fee_account);
+        move_to(fee_account, TransactionFeePreburn<CoinType>{
+            preburn: Libra::new_preburn_with_capability(burn_cap)
```

Comment:
can we maybe get rid of publish_preburn and new_preburn? what's the use of un_approved Preburn resource?

Reply:
I believe we had these for testing purposes in testnet, but I think they can be removed now. @sblackshear do you know of any reason to keep these around? I can't think of any remaining reasons.